### ISSUE TITLE
azure_rm inventory not work for MSI auth with others AzureCloud.

### ISSUE THREAD

**Issue by tu-doan** [ansible/ansible_collections_azure#893]
## Description:
In `azure_rm.yaml`, my config is:
```
---
plugin: azure.azcollection.azure_rm
cloud_environment: AzureChinaCloud
subscription_id: XXXXXX-XXXXXX-XXXX-XXXXX
auth_source: msi
```
Did't know why but ansible playbook return 404 with my subscription. 
```
[WARNING]:  * Failed to parse /runner/inventory/ag1/azure_rm.yml with auto plugin: a batched request failed with status code 404
```

## Debugging

After debugging, I found out that the playbook use AzurePublicCloud endpoint, instead of `AzureChinaCloud `. That's why the response is 404 with my subcribtion_id.

Here is the issue with the library: 
Return `AZURE_PUBLIC_CLOUD` when variable `raw_cloud_env` is None: 
https://github.com/ansible-collections/azure/blob/7b1611da473c0ede88437fb28dc4a02472b01c60/plugins/module_utils/azure_rm_common.py#L1462-L1466

However, when auth_source=msi, the return object is not including `cloud_environment`, while the cli method has:
https://github.com/ansible-collections/azure/blob/7b1611da473c0ede88437fb28dc4a02472b01c60/plugins/module_utils/azure_rm_common.py#L1587-L1625

Despite the authentication via msi is success, these line-of-codes lead the object to none when select attribute cloud_environment. Then with the if clause, it selects the cloud as AzurePublicCloud. After all, other module send batch_request uses the wrong endpoint because of the wrong cloud_environment.

## TL;DR
Module `AzureRMAuth` does not treat `msi` auth method as same as the `cli` method. 
Function `_get_msi_credentials` is missing attribute `cloud_environment` when returns.

**Comment by tu-doan** [comment]
A fixed PR was introduced here: https://github.com/ansible-collections/azure/pull/894
